mon-pix/app/components/routes/register-form.js

```diff
@@ -30,30 +25,51 @@ const isMonthValid = (value) => value > 0 && value <= 12;
 const isYearValid = (value) => value > 999 && value <= 9999;
 const isStringValid = (value) => !!value.trim();
 
-@classic
-export default class RegisterForm extends Component {
-  @inject()
-  session;
+class FormValidation {
+  lastName = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+  firstName = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+  dayOfBirth = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+  monthOfBirth = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+  yearOfBirth = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+  email = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+  password = {
+    @tracked status: 'default',
+    @tracked message: null,
+  }
+}
 
-  @inject()
-  store;
+export default class RegisterForm extends Component {
+  @service session;
+  @service store;
+  @service intl;
 
-  @inject()
-  intl;
+  @tracked isLoading = false;
+  @tracked errorMessage = null;
+  @tracked displayRegisterErrorMessage = false;
+  @tracked matchingStudentFound = false;
+  @tracked loginWithUsername = true;
 
   schoolingRegistrationDependentUser = null;
   schoolingRegistrationUserAssociation = null;
-  isLoading = false;
-  errorMessage = null;
-  displayRegisterErrorMessage = false;
-  matchingStudentFound = false;
-  isPasswordVisible = false;
-  loginWithUsername = true;
-
-  @computed('isPasswordVisible')
-  get passwordInputType() {
-    return this.isPasswordVisible ? 'text' : 'password';
-  }
+  validation = new FormValidation();
```

Comment:
**question:** Est-ce que ça serait pas comme ça au lieu de tracked tous les `lastName.status` and co ? 
```suggestion
  @tracked validation = new FormValidation();
```

Reply:
Il ne manque pas des `@tracked` pour les propriétés de `FormValidation`? D'ailleurs je me demande quel effet a `@tracked` dans des littéraux objets comme ceux-ci — vu que `@tracked` est normalement utilisé à la définition d'une classe :
```
  lastName = {
    @tracked status: 'default',
    @tracked message: null,
  }
```